### PR TITLE
Chore/bump controller runtime

### PR DESCRIPTION
**Pull request by gusfcarvalho**
This fixes controller-runtime bump #1152

### CODE REVIEW

File: pkg/utils/utils_test.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func TestObjectHash(t *testing.T) {
 			input: v1.Secret{Data: map[string][]byte{
 				"xx": []byte("yyy"),
 			}},
-			want: "a9fe13fd43b20829b45f0a93372413dd",
```
Comment (moolen) [inline pkg/utils/utils_test.go:49]: interesting, so a version bump results in different object hash :thinking:

Comment (moolen) [reply pkg/utils/utils_test.go:49]: so all external secrets are then reconciled after the bump

Comment (gusfcarvalho) [reply pkg/utils/utils_test.go:49]: Yes... not sure why though, but this is happening. AFAIK all external secrets are reconciled anyways after an upgrade. Are you thinking some additional issues will happen?

Comment (moolen) [reply pkg/utils/utils_test.go:49]: Yes, they're reconciled, but the hash would prevent the controller from actually syncing it.
I don't think this should be an issue :thinking:

Comment (gusfcarvalho) [reply pkg/utils/utils_test.go:49]: I'll run some manual upgrade tests before finishing this bump...

Comment (gusfcarvalho) [reply pkg/utils/utils_test.go:49]: Not an issue :)

**Review by moolen — APPROVED** [review]
LGTM :smile:

**Comment by gusfcarvalho** [comment]
Closing this until argo project starts using kubernetes 1.24 as well.

**Pull request closed without merging**